src/sentry/tasks/weekly_reports.py

```diff
@@ -547,17 +551,23 @@ def fetch_key_performance_issue_groups(ctx):
         ]
 
 
-# Deliver reports
-# For all users in the organization, we generate the template context for the user, and send the email.
-
-
 def deliver_reports(ctx, dry_run=False, target_user=None, email_override=None):
+    """
+    For all users in the organization, we generate the template context for the user, and send the email.
+    """
     # Specify a sentry user to send this email.
     if email_override:
         target_user_id = (
             target_user.id if target_user else None
         )  # if None, generates report for a user with access to all projects
-        send_email(ctx, target_user_id, dry_run=dry_run, email_override=email_override)
+        template_by_user_id = prepare_template_context(ctx, [target_user_id])
```

Comment:
sounds like this is a list of templates?
can we rename this to better represent the return value? Maybe something like `user_template_context_list`

Reply:
Yeah it's a list of dicts like `[{"context": ..., "user_id": 1}]`